src/ophyd_async/panda/_trigger.py

```diff
@@ -1,15 +1,23 @@
 import asyncio
-from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+from pydantic import BaseModel, Field, field_validator
 
 from ophyd_async.core import TriggerLogic, wait_for_value
-from ophyd_async.panda import SeqBlock, SeqTable, TimeUnits
+from ophyd_async.panda import (
+    PcompBlock,
+    PcompDirectionOptions,
+    SeqBlock,
+    SeqTable,
+    TimeUnits,
+)
 
 
-@dataclass
-class SeqTableInfo:
-    sequence_table: SeqTable
-    repeats: int
-    prescale_as_us: float = 1  # microseconds
+class SeqTableInfo(BaseModel):
+    sequence_table: SeqTable = Field(strict=True)
```

Comment:
Ah, is it to do the checking here? In which case maybe we need to reconsider the use of `TypedDict` for PVA tables and switch completely to pydantic models... We could define a `Table(BaseModel)` with a suitable validator for lengths.

Although, after the conversation we had with @danielballan we could ditch `TypedDict` in favour of a row-wise numpy structured datatype, which would enforce column lengths and avoid the need to pydantic models in PVA structures...

I think this needs discussion in person

Reply:
Yes, I was thinking of bringing up allowing pydantic model classes as signal datatypes.

Reply:
I'm now favouring the row-wise numpy structured datatype. I will raise a separate issue to track, so please can you put it back to `TypedDict` for now and skip the checking of the seq table in the `SeqTableInfo` class?

Reply:
Created #442 to track